tools/test-proxy/Azure.Sdk.Tools.TestProxy/Store/GitStore.cs

```diff
@@ -124,11 +124,37 @@ public async Task<NormalizedString> GetPath(string pathToAssetsJson)
                     string branchGuid = Guid.NewGuid().ToString().Substring(0, 8);
                     string gitUserName = GetGitOwnerName(config);
                     string gitUserEmail = GetGitOwnerEmail(config);
+                    string assetMessage = "Automatic asset update from test-proxy.";
+                    string configurationString = $"-c user.name=\"{gitUserName}\" -c user.email=\"{gitUserEmail}\"";
+
                     GitHandler.Run($"branch {branchGuid}", config);
                     GitHandler.Run($"checkout {branchGuid}", config);
+
+                    /*
+                     * This code works by generating a patch file for SPECIFICALLY the eng folder from main.
+                     * Given that these changes appear as "new" changes, they just look like normal file additions. 
+                     * This totally eliminates the possibility of weird historical merge if main has code that we don't expect. 
+                     * Under azure-sdk-assets, we should never see this, but we have already seen it with specific integration
+                     * test tags under azure-sdk-assets-integration. By keeping it as "patch", the soft RESET on unsuccessful 
+                     * push action will properly put their repo into the expected "ready to push" state that a failed
+                     * merge would NOT.
+                     */
+                    GitHandler.Run($"diff --output=changes.patch --no-color --binary --no-prefix HEAD main -- eng/", config);
```

Comment:
Given main should never have anything under the sdk path maybe just always produce a patch for everything. I'm guessing there is also a way to exclude or revert certain paths if you want to be certain nothing comes under there.

Reply:
Actually I guess that won't work with the sparce checkout though.

Reply:
Yeah. I'm torn. I honestly like the comfort of

> pull eng/ and gitignore and nothing else. It should just work given that main is definitely newer than any of the tags.

We already have a couple test cases that are exercising when there _are no_ changes from `main`, and those work just fine.

I feel like just diffing main will definitely _work_, but then we're back to the 

> but what if

To be clear that's just FUD on my part.

The real benefit that I discovered today is really that the patches from main come in as _part of the changeset_ so there is just a single commit with the users name, versus each tag having an "additional" tag on top of it that is merging main. Meaning that the tag won't properly display the recording changes unless the merge did nothing. Otherwise the tag commit is just a commit with the changes to `eng`, and you'll need to `browse files` to see just the recording changes.

I may swap to just merge everything? I'm trying a couple experiments locally to double check all the functionality regardless.